app/src/app/[lng]/data/[step]/SubsectorDrawer.tsx

```diff
@@ -257,6 +293,60 @@ export function SubsectorDrawer({
               dataQuality: value.activity.dataQuality,
               notes: value.activity.sourceReference,
             };
+
+            // save emissions factors (either values for custom or the IDs for each gas for predefined)
+            const gases = ["CO2", "CH4", "N2O"];
+            inventoryValue.gasValues = gases.flatMap((gas) => {
+              let gasValue: GasValueData = {
+                gas,
+                gasAmount: undefined, // set this to null to remove previous data?
+              };
+              if (value.activity.emissionFactorType === "custom") {
+                const emissionsPerActivity = Number(
+                  (value.activity as unknown as Record<string, string>)[
+                    gas.toLowerCase() + "EmissionFactor"
+                  ],
+                );
+                gasValue.emissionsFactor = {
+                  emissionsPerActivity,
+                  gas,
+                  units: value.activity.activityDataUnit,
+                  inventoryId: inventoryId,
+                };
+              } else {
+                if (!emissionsFactors || !scopes) {
+                  throw new Error(
+                    "Emissions factors or scopes not loaded at save time!",
+                  );
+                }
+                const subCategory = scopes?.find(
+                  (scope) => scope.value === subCategoryId,
+                );
+                const gpcReferenceNumber = subCategory?.gpcReferenceNumber;
+                const factors = emissionsFactors.filter((factor) => {
+                  return (
+                    factor.gas === gas &&
+                    factor.units === value.activity.activityDataUnit &&
+                    factor.gpcReferenceNumber === gpcReferenceNumber &&
+                    determineEmissionsFactorType(factor) ===
+                      value.activity.emissionFactorType
+                  );
+                });
+
+                if (factors.length === 0) {
+                  return [];
+                }
+                if (factors.length > 1) {
+                  logger.warn(
+                    `Multiple emissions factor candidates for scope ${gpcReferenceNumber}, units ${value.activity.activityDataUnit}, type ${value.activity.emissionFactorType}`,
```

Comment:
Do we need to toast this warning, for user using tabs, or any other user?

Reply:
In practice this isn't happening anymore after fixing the filtering logic. So this is mainly relevant for developers in case we add new emissions factors in the future and they have invalid data.